mmv1/templates/terraform/constants/spanner_instance.go.erb

```diff
@@ -0,0 +1,58 @@
+<%# The license inside this block applies to this file
+	# Copyright 2021 Google Inc.
+	# Licensed under the Apache License, Version 2.0 (the "License");
+	# you may not use this file except in compliance with the License.
+	# You may obtain a copy of the License at
+	#
+	#     http://www.apache.org/licenses/LICENSE-2.0
+	#
+	# Unless required by applicable law or agreed to in writing, software
+	# distributed under the License is distributed on an "AS IS" BASIS,
+	# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+	# See the License for the specific language governing permissions and
+	# limitations under the License.
+-%>
+func deleteSpannerBackups(d *schema.ResourceData, config *Config, res map[string]interface{}, userAgent string, billingProject string) error {
+	var v interface{}
+	var ok bool
+
+	v, ok = res["backups"]
+	if !ok || v == nil {
+		return nil
+	}
+
+	switch v.(type) {
```

Comment:
Can you comment above this switch why we're switching? It's not immediately obvious that we may have a single backup returned by the API outside of a list.

Reply:
hmm, is this needed? i borrowed the logic in nested_query.go.erb which is used with resources like google_compute_router_* . backups is a nested field in the API response which is similar to those resources.

Reply:
Only if the API has this behaviour, we can remove it otherwise. The generic `nested_query` code includes it automatically, but since this is handwritten we can remove it if it's not applicable.

Reply:
Removed the switch. API returns a list https://cloud.google.com/spanner/docs/reference/rest/v1/projects.instances.backups/list